server/midddleware/auth.go

```diff
@@ -122,13 +122,15 @@ func AuthFunction(ctx context.Context, jwtValidator *validator.Validator, config
 		if customClaims, ok := validatedClaims.CustomClaims.(*CustomClaim); ok {
 			// if incoming namespace is empty, set it to unknown for observables and reject request
 			if customClaims.Namespace.Code == "" {
+				log.Warn().Msg("Valid token with empty namespace received")
 				ctx = request.SetNamespace(ctx, UnknownNamespace)
 				return ctx, api.Errorf(api.Code_UNAUTHENTICATED, "You are not authorized to perform this admin action")
 			}
 			isAdmin := fullMethodNameFound && request.IsAdminApi(fullMethodName)
 			if isAdmin {
 				// admin api being called, let's check if the user is of admin allowed namespaces
 				if !config.Auth.AdminNamespaces.Contains(customClaims.Namespace.Code) {
+					log.Warn().Msgf("Valid token with namespace=%s received", customClaims.Namespace.Code)
```

Comment:
should we mention in the log that it is not from the allowed list?

Reply:
fixed